ipu.py

```diff
@@ -379,40 +380,42 @@ def boot_iso_with_redfish(self, iso_path: str) -> None:
         logger.info("unsetting boot source override")
         self._unset_bootsource_override()
 
-    def _wait_iso_downloaded(self, iso_path: str) -> None:
-        size = get_size(iso_path)
-        if size is None:
-            logger.error_and_exit(f"Couldn't get size of iso {iso_path}")
-        logger.info(f"Size of {iso_path} is {size}")
+    def _virtual_media_is_inserted(self, filename: str) -> bool:
+        result = self._run_curl(f"-k 'https://{self.url}:8443/redfish/v1/Systems/1/VirtualMedia/1'")
+        if result.returncode != 0:
+            return False
+        try:
+            data = json.loads(result.out)
+        except Exception:
+            return False
+        v_inserted = data.get("Inserted")
+        if not isinstance(v_inserted, bool) or not v_inserted:
+            return False
+        v_imageName = data.get("ImageName")
+        if not isinstance(v_imageName, str) or v_imageName != filename:
+            return False
 
-        rh = host.RemoteHost(self.url)
-        rh.ssh_connect("root", password="", discover_auth=False)
-        loop_count = 0
+        return True
+
+    def _wait_iso_downloaded(self, iso_path: str) -> None:
+        logger.info("Waiting for iso_path {repr(iso_path)} to be inserted as VirtualMedia")
+        wait_until = time.monotonic() + 25 * 60
+        filename = extract_filename(iso_path)
+        sleep_time = 60.0
         while True:
-            result = rh.run("du -b /mnt/imc/acc-os.iso").out.split()
-            if len(result) == 0:
-                continue
-            downloaded_size = rh.run("du -b /mnt/imc/acc-os.iso").out.split()[0]
-            percentage = (int(downloaded_size) / int(size)) * 100
-            if loop_count % 6 == 0:
-                logger.info(f"Downloaded {downloaded_size} of {size} ({percentage:.2f}%)")
-            loop_count += 1
-            if downloaded_size == size:
-                break
-            time.sleep(10)
-        logger.info(f"Downloaded {downloaded_size} of {size} ({percentage:.2f}%)")
```

Comment:
Can we still keep reporting the download status as we are polling?

```
2024-10-18 07:23:33 INFO [th:140146282260032] (ipu.py:398): Downloaded 7864320 of 9453174784 (0.08%)
2024-10-18 07:24:36 INFO [th:140146282260032] (ipu.py:398): Downloaded 1243283456 of 9453174784 (13.15%)
2024-10-18 07:25:38 INFO [th:140146282260032] (ipu.py:398): Downloaded 1997684736 of 9453174784 (21.13%)
2024-10-18 07:26:41 INFO [th:140146282260032] (ipu.py:398): Downloaded 2500001792 of 9453174784 (26.45%)
```

Reply:
`_run_curl()` is very verbose. If such status updates can be noted in the output, then the other logging needs to be reduced.

Done and PR updated.